src/components/questionnaire.js

```diff
@@ -0,0 +1,224 @@
+			// A custom method or the options should figure out what this is
+			var isMethod = typeof activeQuestion.options === 'function';
+
+			if (isMethod) {
+				answerInfo.nextStep = activeQuestion.options(this.state.answers);
+			}
+		}
+
+		// Set the active question to answered
+		this.setLogQuestionToAnswered(questionId);
+
+		// Push the answer to the question log
+		this.addToLog({
+			text: answerInfo.text,
+			type: 'answer'
+		});
+
+		// Add the answer to the answer list
+		this.setAnswer(activeQuestion.key, answerValue);
+
+		// Go to the next question, if it exists
+		this.getNextMessage(answerInfo.nextStep);
+	};
+
+	handleFormSubmit = (e) => {
+		console.log(e);
+
+	};
+	
+	handleTextInputChange = (questionKey, validationTypes, changeEvent) => {
+		const answers = this.state.answers;
+		const zipStr = changeEvent.target.value;
+		const validationErrors = this.getValidationErrors(validationTypes, zipStr, questionKey);
+
+		if (validationErrors.length === 0) {
+			answers[questionKey] = zipStr;
+		}
+		else {
+			answers[questionKey] = null;
+		}
+
+		this.setState({
+			answers: answers
+		});
+	};
+
+    restartQuestionnaire = () => {
+        var initialState = this.getInitialState();
+        this.setState(initialState, () => {
+            this.startQuestionnaire();
+        });
+	};
+
+	setAnswer = (key, answer) => {
+		let answers = this.state.answers;
+		answers[key] = answer;
+
+		this.setState({
+			answers: answers
+		})
+	}
+
+	setLogQuestionToAnswered = (id) => {
+		const messageLog = this.state.log;
+		let logItemToUpdateIndex;
+		let logItemToUpdate;
+
+		for (let i = 0, len = messageLog.length; i < len; i++) {
+			const logItem = Object.assign({}, messageLog[i]);
+			const idExists = Object.hasOwnProperty.call(logItem, 'id');
+			const idMatches = logItem.id === id;
+
+			if (idExists && idMatches) {
+				logItemToUpdate = logItem;
+				logItemToUpdateIndex = i;
+				break;
+			}
+		}
+
+		logItemToUpdate.isAnswered = true;
+		messageLog[logItemToUpdateIndex] = logItemToUpdate;
+
+		this.setState({
+			log: messageLog
+		});
+	};
+	
+    startQuestionnaire() {
+        this.getNextMessage();
+	};
+	
+    render() {
+        const { userInput, disableUserInput } = this.state;
+
+        return (
+            <div className="conversation-container">
+                <section id="conversation" className="conversation">
+                    <section className="conversation-body">
+                        {this.state.isLoading &&
+                            <p>Loading</p>
+                        }
+                        {this.state.log.map((message, index) => {
+                            return <Question 
+										key={index}
+										validationErrors={this.state.validationErrors}
+										message={message}
+										onButtonSelect={this.handleButtonSelect}
+										onQuestionSubmit={this.handleFormSubmit}
+										onHandleTextInputChange={this.handleTextInputChange} />
+						})}
```

Comment:
General comment - when passing through handlers, make sure you know how you want `this` to be treated inside the handler. I've gotten in the habit of binding to the class with `this.myHandler.bind(this)`.

Reply:
moved to the constructor